File: selenium-ruby/todo_test.rb
require 'test/unit'
require 'selenium-webdriver'

class TodoAppTest < Test::Unit::TestCase
  def setup
    # ドライバーにFirefoxを指定
    @driver = Selenium::WebDriver.for :firefox

    # urlを開く
    @driver.navigate.to 'http://localhost:8080/'

    # フォームをサブミットしてチェックボックスをクリック
    @driver.find_element(:css => 'input[type="text"]').send_keys 'foo'
    @driver.find_element(:css => 'form').submit
    @driver.find_element(:css => 'li input[type="checkbox"]').click
  end
 
  def teardown
    @driver.quit
  end
 
  def test_todo
    list = @driver.find_elements :css => 'li'
    item = list[0]

    # 要素の検証
    assert_equal(list.size, 1)
    assert_equal(item.text, 'foo')
    assert_equal(item.attribute('class'), 'complete')
  end
end

File: casper/todo_test.js
var casper = require('casper').create();

// urlを開く
casper.start('http://localhost:8080/');

casper.thenEvaluate(function() {
  // フォームをサブミットしてチェックボックスをクリック
  $('input[type="text"]').val('foo');
  $('form').submit();
  $('li input[type="checkbox"]').click();
});

// 状態の検証
casper.then(function() {
  this.test.assertEvalEquals(function() {
    return $('li').length;
  }, 1, 'Added Todo List');

  this.test.assertEvalEquals(function() {
    return $('li').text();
  }, 'foo', 'Added input value');

  this.test.assertEvalEquals(function() {
    return $('li').attr('class');
  }, 'complete', 'Added complete class');
});

casper.run(function() {
  this.test.done();
  this.test.renderResults(true);
});
